Dikubot/Webapp/Data/FacebookService.cs

```diff
@@ -0,0 +1,139 @@
+            "client_id=" + Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID") +
+            "&redirect_uri="+redirectUri +
+            "&auth_type=rerequest" +
+            "&config_id=1372246313628981";
+
+        _navigationManager.NavigateTo(link);
+    }
+
+
+    public List<FacebookPageModel> AddLongLivedAccessTokenToDatabase(string redirectUri, string codeParameter)
+    {
+        return AddLongLivedAccessTokenToDatabase(GetLongLivedPageAccessToken(redirectUri, codeParameter));
+    }
+    
+    public List<FBPage> GetLongLivedPageAccessToken(string redirectUri, string codeParameter)
+    {
+        try
+            {
+                
+                // Link to get the short lived User AccessToken
+                var shortlivedAccessTokenLink = $"https://graph.facebook.com/{graphApiVersion}/oauth/access_token?client_id={appId}&redirect_uri={redirectUri}&client_secret={appSecret}&code={codeParameter}";
+
+                FBAccessToken shortLivedAccessToken = _httpClient.GetFromJsonAsync<FBAccessToken>(shortlivedAccessTokenLink).Result;
+                
+                // Link to get the short lived User AccessToken
+                var getFacebookUsersPagesLink = $"https://graph.facebook.com/{graphApiVersion}/me/accounts?access_token={shortLivedAccessToken.AccessToken}";
+
+                var _fbPagesList = _httpClient.GetFromJsonAsync<FBRespons<FBPage>>(getFacebookUsersPagesLink).Result.Data;
+
+                // Link to get the Long lived User AccessToken
+                var longLivedAccessTokenLink = $"https://graph.facebook.com/{graphApiVersion}/oauth/access_token?grant_type=fb_exchange_token&" +
+                            $"client_id={appId}&" +
+                            $"client_secret={appSecret}&" +
+                            $"fb_exchange_token={shortLivedAccessToken.AccessToken}";
+                var longLivedUserAccessToken = _httpClient.GetFromJsonAsync<FBAccessToken>(longLivedAccessTokenLink).Result;
+                
+                    
+                // Link to get the unique app User id
+                var userIdLink = $"https://graph.facebook.com/me?fields=id&access_token={longLivedUserAccessToken.AccessToken}";
+                var appScopedUserId = _httpClient.GetFromJsonAsync<FBidRespons>(userIdLink).Result.Id;
+
+                //Link to get Long Lived Page Access Token and page info
+                var longLivedPageAccessTokenLink = $"https://graph.facebook.com/{graphApiVersion}/{appScopedUserId}/" +
+                            $"accounts?access_token={longLivedUserAccessToken.AccessToken}";
+
+                //List of FBPag
+
+                return _httpClient.GetFromJsonAsync<FBRespons<FBPage>>(longLivedPageAccessTokenLink).Result.Data;
+                
+                
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
```

Comment:
This is bad error handling. It would be better to handle the errors and have good and understandable errors passed to the users. I will let it slide in this case, but keep it in mind.